File: tests/java/tests/java/org/elastos/hive/config/ApplicationConfig.java
package org.elastos.hive.config;

import com.fasterxml.jackson.annotation.JsonProperty;

public class ApplicationConfig extends Config {
	@JsonProperty("name")
	private String name;
	@JsonProperty("mnemonic")
	private String mnemonic;
	@JsonProperty("passPhrase")
	private String passPhrase;
	@JsonProperty("storepass")
	private String storepass;

	public String name() {
		return this.name;
	}

	public String mnemonic() {
		return this.mnemonic;
	}

	public String passPhrase() {
		return this.passPhrase;
	}

	public String storepass() {
		return this.storepass;
	}

	public static ApplicationConfig deserialize(String content) {
		return deserialize(content, ApplicationConfig.class);
	}
}

File: src/config/applicationconfig.ts
export class ApplicationConfig {
	
	private name: string;
	private mnemonic: string;
	private passPhrase: string;
	private storepass: string;

	public ApplicationConfig (name: string, mnemonic: string, passPhrase: string, storepass: string) {
		this.name = name;
		this.mnemonic = mnemonic;
		this.passPhrase = passPhrase;
		this.storepass = storepass;
	}

	public getName(): string {
		return this.name;
	}

	public getMnemonic(): string {
		return this.mnemonic;
	}

	public getPassPhrase(): string {
		return this.passPhrase;
	}

	public getStorepass(): string {
		return this.storepass;
	}

	public static deserialize(json: any): ApplicationConfig {
		let jsonObj = (typeof json === "string" ? JSON.parse(json as string) : json);
		let newConfig = new ApplicationConfig();
		newConfig.name = jsonObj['name'];
		newConfig.mnemonic = jsonObj['mnemonic'];
		newConfig.passPhrase = jsonObj['passPhrase'];
		newConfig.storepass = jsonObj['storepass'];

		return newConfig;
	}
}
